inc/abstracttarget.class.php

```diff
@@ -1501,6 +1512,79 @@ protected function showLocationSettings($rand) {
       echo '</tr>';
    }
 
+   protected function showValidationSettings($rand) {
+      echo '<tr>';
+
+      // Setting label
+      echo '<td width="15%">' . __('Validation') . '</td>';
+
+      // Possible values
+      echo '<td width="45%">';
+      Dropdown::showFromArray('validation_rule', static::getEnumValidationRule(), [
+         'value'     => $this->fields['validation_rule'],
+         'on_change' => "plugin_formcreator_change_validation($rand)",
+         'rand'      => $rand
+      ]);
+      echo Html::scriptBlock("plugin_formcreator_change_validation($rand)");
+      echo '</td>';
+
+      // Hidden secondary labels, displayed according to the user main choice
+      echo '<td width="15%">';
+      $titles = [
+         'validation_specific_title'       => __('Approver'),                 // VALIDATION_RULE_SPECIFIC_USER_OR_GROUP
+         'validation_from_question_title'  => __('Question', 'formcreator'),  // VALIDATION_RULE_ANSWER_USER
+      ];
+      foreach ($titles as $id => $title) {
+         echo "<span id='$id' style='display: none'>$title</span>";
+      }
+      echo '</td>';
+
+      // Hidden secondary values, displayed according to the user main choice
+      echo '<td width="25%">';
+
+      // VALIDATION_RULE_SPECIFIC_USER_OR_GROUP
+      echo '<div id="validation_specific" style="display: none">';
+      $validation_dropdown_params = [
+         'name' => 'validation_specific'
+      ];
+      $validation_data = json_decode($this->fields['validation_question'], true);
+      if (!is_null($validation_data) && isset($validation_data['type'])) {
+         $validation_dropdown_params['users_id_validate'] = $validation_data['values'];
+      }
+      CommonITILValidation::dropdownValidator($validation_dropdown_params);
+      echo '</div>';
+
+      // VALIDATION_RULE_ANSWER_USER
+      echo '<div id="validation_answer_user" style="display: none">';
+      $question = new PluginFormcreatorQuestion();
+      $form_questions = $question->getQuestionsFromForm($this->getForm()->getID(), [
+         'fieldtype' => ['actor', 'glpiselect']
+      ]);
+      $user_questions = [];
+      foreach ($form_questions as $id => $form_question) {
+         if ($form_question->fields['fieldtype'] == 'actor') {
+            // Get all "actor" questions
+            $user_questions[$id] = $form_question->fields['name'];
+         } else if ($form_question->fields['fieldtype'] == 'glpiselect') {
+             // Get all "glpiselect" question on users
+            $decoded_values = json_decode(
+               $form_question->fields['values'],
+               JSON_OBJECT_AS_ARRAY
+            );
+            if (($decoded_values['itemtype'] ?? "") === 'User') {
+               $user_questions[$id] = $form_question->fields['name'];
+            }
+         }
+      }
+      Dropdown::showFromArray('_validation_from_user_question', $user_questions, [
+         'value' => $this->fields['validation_question'],
+      ]);
```

Comment:
PluginFormcreatorQueston::dropdownForForm() should help to replace this big snippet of code.

The following should work, worth to test it and adjust if necessary.

```suggestion
      PluginFormcreatorQuestion::dropdownForForm(
         $this->getForm()->getID(),
         [
            new QueryExpression("`fieldtype` = 'actor' OR `fieldtype` = 'glpiselect' AND `itemtype`='User'"),
         ],
         'value' => $this->fields['validation_question'],
      );
```

Reply:
`itemtype` is a JSON property of the `values` field.
I don't think it can be handled with a simple SQL criteria (`fieldtype` = 'glpiselect' AND `itemtype`='User').

Reply:
This no longer true. I changed this recently to solve similar need elsewhere in the plugin.

  https://github.com/pluginsGLPI/formcreator/blob/59ea8b79f00e2ca9017754489f70a08f3406eed6/inc/field/dropdownfield.class.php#L891 

https://github.com/pluginsGLPI/formcreator/blob/59ea8b79f00e2ca9017754489f70a08f3406eed6/install/upgrade_to_2.13.php#L69

Reply:
Ok good to know, done.